docs/proposals/20220712-cluster-api-addon-orchestration.md

```diff
@@ -0,0 +1,368 @@
+- `version`: The version of the Helm chart.
+- `values`: A map of key/value pairs specifying values to be passed to the Helm chart. Go templating is supported to reference fields from each workload cluster spec.
+
+##### Status
+
+- `matchingClusters`: A list of object references to workload clusters matching the cluster selector.
+- `ready`: A boolean that is true when the list of HelmReleaseProxies are up to date.
+- `failureReason`: A string pointer indicating the error message, if applicable.
+
+```yaml
+apiVersion: addons.cluster.x-k8s.io/v1beta1
+kind: HelmChartProxy
+metadata:
+  name: cloud-provider-azure-helmchartproxy
+spec:
+  clusterSelector:
+    key: cloudProviderAzureChart
+    value: enabled
+  repoURL: https://raw.githubusercontent.com/kubernetes-sigs/cloud-provider-azure/master/helm/repo
+  chartName: cloud-provider-azure
+  version: 1.23.8
+  releaseName: cloud-provider-azure
+  values:
+    infra.clusterName: "{{ .Cluster.Name }}"
+    cloudControllerManager.containerResourceManagement.limitsCPU: "{{ add .ControlPlane.Spec.Replicas 10 }}"
+status:
+  matchingClusters:
+  - apiVersion: cluster.x-k8s.io/v1beta1
+    kind: Cluster
+    name: default-15210
+    namespace: default
+  ready: true
+```
+
+#### HelmReleaseProxy CRD
+
+The second CRD, HelmReleaseProxy, is responsible for installing and orchestrating the Helm release on a selected workload Cluster. The HelmReleaseProxy is created as part of the parent HelmChartProxy flow. The parent HelmChartProxy will be responsible for keeping its HelmReleaseProxy children up-to-date as its lifecycle is updated (for example, if the `values` properties of a HelmChartProxy are updated, its child HelmReleaseProxy resources will be updated as well).
+
+##### Spec
+
+- `clusterRef`: An object reference to the Cluster the Helm release is installed on.
+- `repoURL`: The URL of the Helm chart repository.
+- `chartName`: The name of the Helm chart.
+- `releaseName`: The name of the Helm release.
+- `version`: The version of the Helm chart.
+- `values`: A map of key/value pairs specifying values to be used when installing the Helm release. Note that this map is populated with the resolved values from the Cluster.
+
+##### Status
+
+- `ready`: A boolean that indicates when the Helm release is successfully installed and up to date with the HelmReleaseProxy spec.
+- `failureReason`: A string pointer indicating the error message, if applicable.
+
+```yaml
+apiVersion: addons.cluster.x-k8s.io/v1beta1
+kind: HelmReleaseProxy
+metadata:
+  labels:
+    addons.cluster.x-k8s.io/helmchartproxy-name: cloud-provider-azure-chart
+    cluster.x-k8s.io/cluster-name: default-15210
+  name: cloud-provider-azure-default-15210
+  namespace: default
+  ownerReferences:
+  - apiVersion: addons.cluster.x-k8s.io/v1beta1
+    blockOwnerDeletion: true
+    controller: true
+    kind: HelmChartProxy
+    name: cloud-provider-azure-chart
+    uid: <some-uid>
+spec:
+  chartName: cloud-provider-azure
```

Comment:
consider adding another addon that isn't a cloud provider as an example for illustration purposes so people can picture different use cases while reviewing this

Reply:
Sounds good, I'll work on putting together a different example and adding it back.